### PR TITLE
Remove unnecessary vtable for Segment in PageAllocator

### PR DESCRIPTION
**Pull request by curtisman**

### CODE REVIEW

File: lib/Common/Memory/PageAllocator.h [hunk 1]
```diff
@@ -136,13 +139,14 @@ class SegmentBaseCommon
...
+    PREVENT_STANDALONE_HEAPINSTANCE();
 public:
     SegmentBase(PageAllocatorBase<TVirtualAlloc> * allocator, DECLSPEC_GUARD_OVERFLOW size_t pageCount, bool enableWriteBarrier);
-    virtual ~SegmentBase();
+    ~SegmentBase();
 
     size_t GetPageCount() const { return segmentPageCount; }
 
-    // Some pages are reserved upfront for secondary allocations
+    // Some pages are reser0x0000016bf7dd6440ed upfront for secondary allocations
```
Comment (leirocks) [inline lib/Common/Memory/PageAllocator.h]: accidental paste

Comment (curtisman) [reply lib/Common/Memory/PageAllocator.h]: Yep.

**Comment by leirocks** [comment]
@dotnet-bot test OSX

**Comment by curtisman** [comment]
@dotnet-bot 
test OSX _no_jit_shared_osx_osx_test
test OSX static_osx_osx_debug
test OSX static_osx_osx_test

**Comment by curtisman** [comment]
@dotnet-bot test OSX static_osx_osx_debug

**Comment by curtisman** [comment]
@dotnet-bot test OSX static_osx_osx_debug

**Comment by curtisman** [comment]
@dotnet-bot test OSX static_osx_osx_debug

**Pull request merged**